onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -174,13 +181,41 @@ def optimize_by_onnxruntime(
         else:
             providers.append("CUDAExecutionProvider")
 
-    onnxruntime.InferenceSession(onnx_model_path, sess_options, providers=providers, **kwargs)
+    if isinstance(onnx_model, str):
+        onnxruntime.InferenceSession(onnx_model, sess_options, providers=providers, **kwargs)
+    elif isinstance(onnx_model, ModelProto):
+        _load_infer_session_from_modelproto(onnx_model, save_as_external_data, sess_options, providers, kwargs)
 
     assert os.path.exists(optimized_model_path) and os.path.isfile(optimized_model_path)
     logger.debug("Save optimized model by onnxruntime to %s", optimized_model_path)
     return optimized_model_path
 
 
+def _load_infer_session_from_modelproto(model, has_external_data, sess_options, providers, kwargs) -> None:
+    import onnxruntime
+    from onnxruntime_inference_collection import OrtValue
+    from fusion_utils import NumpyHelper
+
+    external_data = []
+    for tensor in model.graph.initializer:
+        name = tensor.name
+
+        if has_external_data:
+            raise ValueError("Model has external data, model path is required to load the inference session.")
+
+        logger.info("externalizing tensor: %s", name)
+        if tensor.HasField("raw_data"):
+            npt = NumpyHelper.to_array(tensor)
+            orv = OrtValue.ortvalue_from_numpy(npt)
+            external_data.append((name, orv))
+            tensor.name = name
+            tensor.ClearField("raw_data")
+
+    external_names, external_values = zip(*external_data)
```

Comment:
Is there a good way to unit test this part of the logic?

Reply:
Honestly I was trying to add a unit test for this, but there is no unit test for this file.

Reply:
You can add a tiny model with raw_data, and test it like
https://github.com/microsoft/onnxruntime/blob/15219e2e71b82f906f92317510396b6ccc858c49/onnxruntime/test/python/transformers/test_optimizer.py#L195